### ISSUE TITLE
Reference scheme reverts to http when using https

### ISSUE THREAD

**Issue by stephencpope** [restlet/restlet-framework-java#998]
Originally reported on stackoverflow.com: http://stackoverflow.com/questions/27663181/restlet-reference-scheme-reverts-to-http-when-using-https

I have a simple Restlet application using the embedded org.restlet.Server in restlet-jse-2.2.3 which worked fine with HTTP. Following the Restlet book, I converted it to use HTTPS instead. However, within my ServerResource instances, getReference() is returning References using the HTTP scheme even though the application is only accessed vi HTTPS. Even the org.restlet.routing.Redirector is confused, taking a relative target and returning a URI with the http schema.

Here's the gist of my main():
```java
    Component component = new Component();
    Server server = component.getServers().add(Protocol.HTTPS, 8081);
    Series<Parameter> parameters = server.getContext().getParameters();
    parameters.add("keystorePath", ...);
    parameters.add("keystorePassword", ...);
    parameters.add("keystoreType", ...);
    parameters.add("keyPassword", ...);
    component.getDefaultHost().attachDefault(new MyApplication());
    component.start();
```
 And in MyApplication.createInboundRoot():
```java
    Router router = new Router(getContext());
    router.attach("/", new Redirector(getContext(), "/servers",
                                                        Redirector.MODE_CLIENT_FOUND));
    router.attach(...);
    ...
```
 All my various attached ServerResource instances work fine when addressed as `https://my.host:MYPORT/...`, but the Reference instances returned by their getReference() method is always of the form `http://my.host:MYPORT/...` forcing me to override the scheme to https to generate relative links, and hitting `https://my.host:MYPORT/` returns a redirect to `http://my.host:MYPORT/servers`, which of course then fails.
 
I did some more digging on this. I do not claim a deep understanding of the Restlet implementation, but it appears to me that the problem is that the org.restlet.engine.connector.HttpsServerHelper.confidential flag is not being set to true and that similarly when an org.restlet.engine.connector.HttpExchangeCall instance is created by the HttpsServerHelper, its confidential flag is also not being set. This results in creating the org.restlet.engine.adapter.HttpRequest instance with both the resourceRef and originalRef references initialized with the http schema instead of https. In the meanwhile, I was able to force the issue by adding the following handle() implementation to the MyApplication class:

```java
  public void handle(Request request, Response response) {
    request.getResourceRef().getBaseRef().setScheme("https");
    request.getResourceRef().setScheme("https");
    request.getOriginalRef().setScheme("https");
    super.handle(request, response);
  }
```
which is crude but gets the job done.

**Comment by thboileau** [comment]
Hi, thanks a lot for pointing this issue which is clearly a bug.
I think also the error is in the org.restlet.engine.connector.HttpsServerHelper class in the "start" method.

**Comment by thboileau** [comment]
Thanks a lot Stephen, the fix is available in the 2.3 and master branches.

**Comment by stephencpope** [comment]
My pleasure, Thierry. As I mentioned I have a workaround at the application level so it is not impeding my own progress at this point although it would be great to have it fixed in an upcoming point release!

Stephen
________________________________________
